packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  FoldedCollection,
+  FoldedResult,
+  FoldedResultList,
+  FoldedResultListState,
+  Result,
+  ResultList,
+  ResultListState,
+  ResultsPerPageState,
+  ResultTemplatesManager,
+  ResultTemplate,
+  buildResultTemplatesManager,
+  ResultListProps,
+  EcommerceDefaultFieldsToInclude,
+} from '@coveo/headless';
+import {identity} from 'lodash';
+import {Bindings} from '../../utils/initialization-utils';
+import {
+  ResultDisplayDensity,
+  ResultDisplayImageSize,
+  ResultDisplayLayout,
+} from '../atomic-result/atomic-result-display-options';
+import {TemplateContent} from '../result-templates/result-template-common';
+
+interface DisplayProps {
+  density: ResultDisplayDensity;
+  imageSize?: ResultDisplayImageSize;
+  image?: ResultDisplayImageSize;
+  display?: ResultDisplayLayout;
+}
+export interface ResultPlaceholderProps extends DisplayProps {
+  resultsPerPageState: ResultsPerPageState;
+}
+export interface ResultsProps extends DisplayProps {
+  resultListState: FoldedResultListState | ResultListState;
+  bindings: Bindings;
+  host: HTMLElement;
+  getContentOfResultTemplate(
+    result: Result | FoldedResult
+  ): HTMLElement | DocumentFragment;
+
+  classes: string;
+}
+
+export interface AtomicResultListBase extends Omit<ResultsProps, 'classes'> {
+  templateHasError: boolean;
+  listWrapperRef?: HTMLDivElement;
+  resultsPerPageState: ResultsPerPageState;
+
+  resultListCommon?: ResultListCommon;
+
+  resultList: FoldedResultList | ResultList;
+}
+
+export function getId(
+  result: Result | FoldedCollection,
+  resultListState: FoldedResultListState | ResultListState
+) {
+  return unfolded(result).uniqueId + resultListState.searchResponseId;
+}
+
+export function unfolded(result: Result | FoldedResult): Result {
+  return (result as FoldedResult).result ?? result;
+}
+
+export function getFields(fieldsToInclude: string): string[] {
```

Comment:
I would rename this to something like `determineFieldsToInclude` and rename the parameter to something like `configuredFieldsToInclude` or something similar